browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchAdapter.java

```diff
@@ -0,0 +1,109 @@
+package com.braintreepayments.browserswitch;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+
+public class BrowserSwitchAdapter implements IBrowserSwitchClient {
+
+    static BrowserSwitchAdapter newInstance() {
```

Comment:
Once we do get rid of `BrowserSwitchFragment`, is the plan to also get rid of `BrowserSwitchAdapter` and consolidate to `BrowserSwitchClient`?

Reply:
We can technically but I think the adapter pattern works well here because it allows BrowserSwitchClient to only have to consider one method internally. The adapter here allows us to add additional overloads more easily as the API evolves along with the Android SDK.

Reply:
This was also another component that made the test for `start()` a bit easier in [BrowserSwitchClient](https://github.com/braintree/browser-switch-android/blob/0618df674f1aee42823408032dd5f32a0e09636f/browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchClient.java#L30).

We might be able to achieve something similar to iOS in the future by having an `IBraintreeClientInternal` interface that extends from `IBraintreeClient` and draws a clean line between internal methods/properties and what we want visible to merchants.

Reply:
Gotcha, thanks for explaining!